types/rational.go

```diff
@@ -109,12 +109,14 @@ func (r Rat) Num() int64        { return r.Rat.Num().Int64() }   // Num - return
 func (r Rat) Denom() int64      { return r.Rat.Denom().Int64() } // Denom  - return the denominator
 func (r Rat) IsZero() bool      { return r.Num() == 0 }          // IsZero - Is the Rat equal to zero
 func (r Rat) Equal(r2 Rat) bool { return (&(r.Rat)).Cmp(&(r2.Rat)) == 0 }
-func (r Rat) GT(r2 Rat) bool    { return (&(r.Rat)).Cmp(&(r2.Rat)) == 1 }              // greater than
-func (r Rat) LT(r2 Rat) bool    { return (&(r.Rat)).Cmp(&(r2.Rat)) == -1 }             // less than
-func (r Rat) Mul(r2 Rat) Rat    { return Rat{*new(big.Rat).Mul(&(r.Rat), &(r2.Rat))} } // Mul - multiplication
-func (r Rat) Quo(r2 Rat) Rat    { return Rat{*new(big.Rat).Quo(&(r.Rat), &(r2.Rat))} } // Quo - quotient
-func (r Rat) Add(r2 Rat) Rat    { return Rat{*new(big.Rat).Add(&(r.Rat), &(r2.Rat))} } // Add - addition
-func (r Rat) Sub(r2 Rat) Rat    { return Rat{*new(big.Rat).Sub(&(r.Rat), &(r2.Rat))} } // Sub - subtraction
+func (r Rat) GT(r2 Rat) bool    { return (&(r.Rat)).Cmp(&(r2.Rat)) == 1 }                   // greater than
+func (r Rat) GTE(r2 Rat) bool   { return ((&(r.Rat)).Cmp(&(r2.Rat)) == 1 || r.Equal(r2)) }  // greater than or equal
```

Comment:
Slightly more efficient - `r.GTE(r2) == !r2.GT(r)`

Reply:
That's actually incorrect because of the equals case however, can do `!r.LT(r2)`

Reply:
if `r2 == r`, `r2.GT(r)` is `false`, so `!r2.GT(r)` is true? What case do you mean?

`!r.LT(r2)` == `!r2.GT(r)`, it's just `!(a < b) == !(b > a)`

Reply:
lol oh yeah `!(r < r2)` is the same as my suggestion`!(r2 > r)` just confused myself - you're correct - I think keeping the order left to right, r to r2 is more straight forward though, so my preference is still using `!r.LT(r2)` but just being nit-picky

Reply:
addressed